### PR TITLE
Add S3 tagging tests

### PR DESCRIPTION
**Pull request by ZhuTopher**
### What changes are proposed in this pull request?

Adds integration testing for tagging functionality in the S3 REST API handler.

### Why are the changes needed?

Test coverage for the S3 REST API.

### Does this PR introduce any user facing changes?

- No user-facing changes

### CODE REVIEW

File: core/server/proxy/src/main/java/alluxio/proxy/s3/TaggingData.java [hunk 1]
```diff
@@ -23,18 +28,28 @@
...
+import javax.annotation.Nullable;
+import javax.xml.bind.annotation.XmlTransient;
 
 /**
  * Parsed version of the S3 tagging XML.
  * Used by the following S3 API actions:
  * - GetBucketTagging, GetObjectTagging, PutBucketTagging, PutObjectTagging
  */
 @JacksonXmlRootElement(localName = "Tagging")
+@JsonIgnoreProperties(ignoreUnknown = true, value = {"tagMap"})
```
Comment (ZhuTopher) [inline core/server/proxy/src/main/java/alluxio/proxy/s3/TaggingData.java]: I was unable to marshal this struct while ignoring the metadata field `mTagMap` without specifying the rendered name `tagMap` in this annotation.

I'm not sure why `@XmlTransient` was insufficient, nor do I know if I have redundant/no-op annotations at this point.

File: tests/src/test/java/alluxio/client/rest/S3ClientRestApiTest.java [hunk 2]
```diff
@@ -978,16 +1036,38 @@ private HttpURLConnection deleteBucketRestCall(String bucketUri) throws Exceptio
...
+  private void createObjectRestCall(String objectUri, byte[] objectContent,
+                                    Map<String, String> params,
+                                    Map<String, String> extraHeaders) throws Exception {
+    createObjectRestCall(objectUri, objectContent, params,
+        computeObjectChecksum(objectContent), extraHeaders);
+  }
+
+  private void createObjectRestCall(String objectUri, byte[] objectContent,
+                                    Map<String, String> params,
+                                    String md5, Map<String, String> extraHeaders) throws Exception {
```
Comment (ZhuTopher) [inline tests/src/test/java/alluxio/client/rest/S3ClientRestApiTest.java]: Java-native "optional" parameters via method overloading as opposed to callers specifying `null`. If this is too verbose then I can use a single method with two null parameter checks like previously.

Comment (ZhuTopher) [reply tests/src/test/java/alluxio/client/rest/S3ClientRestApiTest.java]: I decided to axe the method overloading for the verbosity mentioned above.

File: core/server/proxy/src/main/java/alluxio/proxy/s3/TaggingData.java [hunk 3]
```diff
@@ -11,61 +11,55 @@
...
+import javax.xml.bind.annotation.XmlTransient;
 
 /**
  * Parsed version of the S3 tagging XML.
  * Used by the following S3 API actions:
  * - GetBucketTagging, GetObjectTagging, PutBucketTagging, PutObjectTagging
  */
 @JacksonXmlRootElement(localName = "Tagging")
-public class TaggingData implements Serializable {
-  public static final long serialVersionUID = 1L;
```
Comment (ZhuTopher) [inline core/server/proxy/src/main/java/alluxio/proxy/s3/TaggingData.java:37]: I recently learned that I wasn't serializing this as an XML, but rather as a plain Java object (d'oh). So I switched the implementations in this PR.

File: core/server/proxy/src/main/java/alluxio/proxy/s3/TaggingData.java [hunk 4]
```diff
@@ -111,12 +110,69 @@ public String toString() {
...
+   * Fills the tag map with the contents of the tag set.
+   */
+  @XmlTransient
+  private void repopulateTagMap() {
+    mTagMap.clear();
+    for (TagObject tag : mTagSet.mTags) {
+      mTagMap.put(tag.mKey, tag.mValue);
+    }
+  }
+
```
Comment (ZhuTopher) [inline core/server/proxy/src/main/java/alluxio/proxy/s3/TaggingData.java]: I added a separate data structure to mirror the underlying XML structure for convenience when trying to iterate on/modify said data.

**Review by yuzhu — APPROVED** [review]

**Comment by yuzhu** [comment]
LGTM if test passes

**Pull request merged**